src/components/FeedComponent.ts

```diff
@@ -0,0 +1,113 @@
+import { Post } from "../models/Post";
+import { AuthService } from "../services/AuthService";
+import { FeedService } from "../services/FeedService";
+
+export class FeedComponent {
+  static showFeed(): void {
+    const feedContainer = document.getElementById("feed-container")!;
+    feedContainer.innerHTML = "";
+
+    const currentUser = AuthService.getCurrentUser();
+    const posts = FeedService.getPosts();
+
+    if (posts.length === 0) {
+      feedContainer.innerHTML = "<p>No posts available.</p>";
+    } else {
+      posts.forEach((post) => {
+        const postElement = document.createElement("div");
```

Comment:
This section is too large, please divide it into different private functions to create the different elements needed for a post, create buttons, etc. Also, you can think of something like a builder or Factory pattern or just doing it on separate functions.